Source/FixedBackground/BoostedBHFixedBG.hpp

```diff
@@ -0,0 +1,244 @@
+                (dAdx[j] / A - (dBdx[j] - dAdx[j] * v2) / (B - A * v2));
+        }
+
+        // calculate the extrinsic curvature, using the fact that
+        // 2 * lapse * K_ij = D_i \beta_j + D_j \beta_i - dgamma_ij dt
+        // and dgamma_ij dt = 0 in chosen fixed gauge
+        const auto chris_phys = compute_christoffel(vars.d1_gamma, gamma_UU);
+        FOR2(i, j)
+        {
+            vars.K_tensor[i][j] = 0.0;
+            FOR1(k)
+            {
+                vars.K_tensor[i][j] +=
+                    vars.gamma[k][j] * vars.d1_shift[k][i] +
+                    vars.gamma[k][i] * vars.d1_shift[k][j] +
+                    (vars.d1_gamma[k][i][j] + vars.d1_gamma[k][j][i]) *
+                        vars.shift[k];
+                FOR1(m)
+                {
+                    vars.K_tensor[i][j] += -2.0 * chris_phys.ULL[k][i][j] *
+                                           vars.gamma[k][m] * vars.shift[m];
+                }
+            }
+            vars.K_tensor[i][j] *= 0.5 / vars.lapse;
+        }
+        vars.K = compute_trace(gamma_UU, vars.K_tensor);
+    }
+
+  protected:
+    /// Work out the gradients of the quantities H, A and B appearing in the
+    /// Isotropic Schwarzschild solution
+    template <class data_t>
+    void get_metric_derivs(Tensor<1, data_t> &dAdx, Tensor<1, data_t> &dBdx,
+                           const Coordinates<data_t> &coords) const
+    {
+        // black hole params - mass M and boost v
+        const double M = m_params.mass;
+        const double v = m_params.velocity;
+        const double boost = pow(1 - v * v, -0.5);
+
+        // work out where we are on the grid including effect of boost
+        // on x direction (length contraction)
+        Tensor<1, data_t> x;
+        x[0] = coords.x;
+        x[1] = coords.y;
+        x[2] = coords.z;
+        const data_t x_p = coords.x * boost;
+
+        // the coordinate radius (boosted), subject to a floor
+        const data_t r2 = x_p * x_p + x[1] * x[1] + x[2] * x[2];
+        const data_t r = sqrt(r2);
+
+        // find the H, A, B metric quantities
+        const data_t H = 0.5 * M / r;
+        const data_t sqrtA = (1.0 - H) / (1.0 + H);
+        const data_t A = sqrtA * sqrtA;
+        const data_t B = pow(1.0 + H, 4.0);
+
+        using namespace TensorAlgebra;
+        // derivatives of r wrt actual grid coords
+        Tensor<1, data_t> drdx;
+        FOR1(i) { drdx[i] = x[i] / r; }
+        drdx[0] *= boost * boost;
+
+        // derivs of quantities
+        Tensor<1, data_t> dHdx;
+        FOR1(i)
+        {
+            dHdx[i] = -H / r * drdx[i];
+            dAdx[i] = -4.0 * pow(1.0 + H, -3.0) * (1.0 - H) * dHdx[i];
+            dBdx[i] = 4.0 * pow(1.0 + H, 3.0) * dHdx[i];
+        }
+    }
+
+  public:
+    // used to decide when to excise - ie when within the horizon of the BH
+    // note that this is not templated over data_t
+    double excise(const Cell<double> &current_cell) const
```

Comment:
Maybe change how this works - what should the return value be? An actual distance or a bool?